File: Algorithms/Strings/Super Reduced String.py
import os

#
# Complete the 'superReducedString' function below.
#
# The function is expected to return a STRING.
# The function accepts STRING s as parameter.
#

def superReducedString(s):
    # Write your code here
    stack = []

    for char in s:
        if stack and stack[-1] == char:
            stack.pop()
        else:
            stack.append(char)

    if not stack:
        return "Empty String"
    else:
        return "".join(stack)


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = superReducedString(s)

    fptr.write(result + '\n')
    fptr.close()

File: Algorithms/Strings/Super Reduced String.php
<?php

/*
 * Complete the 'superReducedString' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts STRING s as parameter.
 */

function superReducedString($s) {
    # Write your code here
    $stack = [];

    foreach (str_split($s) as $char) {
        if (!empty($stack) && end($stack) == $char) {
            array_pop($stack);
        }
        else {
            array_push($stack, $char);
        }
    }

    $reducedString = implode('', $stack);

    return $reducedString === '' ? 'Empty String' : $reducedString;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$s = rtrim(fgets(STDIN), "\r\n");
$result = superReducedString($s);

fwrite($fptr, $result . "\n");
fclose($fptr);
